library/common/jni/jni_interface.cc

```diff
@@ -31,20 +32,41 @@ extern "C" JNIEXPORT jlong JNICALL Java_io_envoyproxy_envoymobile_engine_JniLibr
   return init_engine();
 }
 
+static JNIEnv* env() {
```

Comment:
https://stackoverflow.com/questions/27072459/how-to-get-the-message-from-a-java-exception-caught-in-jni